### ISSUE TITLE
Decoding into config appears wrong

### ISSUE THREAD

**Issue by schmidtw** [xmidt-org/sallust#25]
I am trying to configure the encoder for development output and it appears the conversion of the encoder types isn't working.  Here is a simple program showing the problem:

```go
package main

import (
	"bytes"

	"github.com/k0kubun/pp/v3"
	"github.com/spf13/viper"
	"github.com/xmidt-org/arrange"
	"github.com/xmidt-org/sallust"
)

var example = []byte(`
# Sallust Configuration
logging:
  # See: https://github.com/xmidt-org/sallust/issues/24 for details.
  level: debug
  development: true
  encoding: console
  encoderConfig:
    messageKey:       msg
    levelKey:         lvl
    timeKey:          time
    nameKey:          name
    callerKey:        caller
    functionKey:      fn
    stacktraceKey:    stack
    lineEnding:       \n\n
    levelEncoder:     capital
    timeEncoder:      millis
    durationEncoder:  ms
    callerEncoder:    short
    consoleSeparator: ','
  rotation:
    maxage: 120
`)

func main() {
	v := viper.New()

	v.SetConfigType("yaml")
	v.ReadConfig(bytes.NewBuffer(example))

	var c sallust.Config
	err := v.UnmarshalKey("logging", &c, arrange.ComposeDecodeHooks(sallust.DecodeHook))
	if err != nil {
		panic(err)
	}

	pp.Print(c)
}
```
What I get out includes most of what I expect, but none of the `EncodeXXX` fields.  This becomes a problem because I can't then construct a logger based on this configuration correctly.
```
sallust.Config{
  Level:             "debug",
  Development:       true,
  DisableCaller:     false,
  DisableStacktrace: false,
  Sampling:          (*zap.SamplingConfig)(nil),
  Encoding:          "console",
  EncoderConfig:     sallust.EncoderConfig{
    DisableDefaultKeys: false,
    MessageKey:         "msg",
    LevelKey:           "lvl",
    TimeKey:            "time",
    NameKey:            "name",
    CallerKey:          "caller",
    FunctionKey:        "fn",
    StacktraceKey:      "stack",
    LineEnding:         "\\n\\n",
    EncodeLevel:        "",          <------ Missing, expected "capital"
    EncodeTime:         "",          <------ Missing, expected "millis"
    EncodeDuration:     "",          <------ Missing, expected "ms"
    EncodeCaller:       "",          <------ Missing, expected "short"
    EncodeName:         "",
    ConsoleSeparator:   ",",
  },
  OutputPaths:          []string(nil),
  ErrorOutputPaths:     []string(nil),
  InitialFields:        map[string]interface {}{},
  DisablePathExpansion: false,
  Mapping:              func(string) string {...},
  Rotation:             &sallust.Rotation{
    MaxSize:    0,
    MaxAge:     120,
    MaxBackups: 0,
    LocalTime:  false,
    Compress:   false,
  },
}
```

This is using `v0.1.5`.

**Comment by johnabass** [comment]
It looks as though you're just marshaling and then unmarshaling the `sallust.Config` type.  Where is the error when creating the logger via `Config.Build()`?

**Comment by schmidtw** [comment]
There is not error in `Config.Build()`, but the unmarshaling is the problem.  Code that follows this pattern [from argus](https://github.com/xmidt-org/argus/blob/e5260ddc1efec9ee7adcf9aac99a2ee84007a330/setup.go#L57)

```go
	var c sallust.Config
	err = v.UnmarshalKey("logging", &c, arrange.ComposeDecodeHooks(sallust.DecodeHook))
	if err != nil {
		return v, l, err
	}

	l, err = c.Build()
```


will result in a logger that doesn't honor values for the encoders - even though the rest of the configuration values are used.

**Comment by schmidtw** [comment]
I ran into this bit from viper: https://github.com/spf13/viper#unmarshaling ... it looks like it uses the tag `mapstructure` instead of `json` or `yaml`.

**Comment by schmidtw** [comment]
Fixed in release 0.1.6.
